analysis_options.yaml

```diff
@@ -10,45 +10,65 @@ analyzer:
     - "test/goldens/generated_build_script.dart"
 linter:
   rules:
-    # Errors
-    - avoid_empty_else
-    - await_only_futures
-    - comment_references
-    - control_flow_in_finally
-    - empty_statements
-    - hash_and_equals
-    - iterable_contains_unrelated_type
-    - no_duplicate_case_values
-    - test_types_in_equals
-    - throw_in_finally
-    - unawaited_futures
-    - unnecessary_statements
-    - unrelated_type_equality_checks
-    - valid_regexps
-
-    # Style
     - annotate_overrides
+    - avoid_empty_else
     - avoid_function_literals_in_foreach_calls
     - avoid_init_to_null
+    - avoid_null_checks_in_equality_operators
     - avoid_return_types_on_setters
     - avoid_returning_null
     - avoid_unused_constructor_parameters
+    - await_only_futures
     - camel_case_types
+    #- cancel_subscriptions
+    #- cascade_invocations
+    - comment_references
+    #- constant_identifier_names
+    - control_flow_in_finally
     - directives_ordering
     - empty_catches
     - empty_constructor_bodies
+    - empty_statements
+    - hash_and_equals
+    #- implementation_imports
+    - invariant_booleans
```

Comment:
I thought this one slowed things way down

Reply:
I'm not sure...

@pq @bwilkerson – ideas on `invariant_booleans` lint – RE speed?

Reply:
I don't know what the performance characteristics are, at this point. I seem to recall that @pq has a script that will measure the performance, so we might be able to get that information.

Reply:
Yes.  We now have a "benchmark" bot.  This doesn't give us absolute metrics but does get us relative timings.  Here's one run for example:

https://travis-ci.org/dart-lang/linter/jobs/354257315

which confirms that `invariant_booleans` is the slowest of the bunch (by a long shot).

![image](https://user-images.githubusercontent.com/67586/37544736-440e428a-2923-11e8-9c47-62370d19ca65.png)

Whether it's worth it or not is another question altogether!